x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/90_semantic_text_highlighter_bwc.yml

```
@@ -649,5 +649,156 @@ setup:
   - length: { hits.hits.0.highlight.bbq_hnsw_field: 1 }
   - match: { hits.hits.0.highlight.bbq_hnsw_field.0: "ElasticSearch is an open source, distributed, RESTful, search engine which is built on top of Lucene internally and enjoys all the features it provides." }
 
+---
+"Highlighting with type:bbq_disk index options as well as knn with similarity":
+  - requires:
+      cluster_features: "semantic_text.highlighter.bbq_and_similarity_support"
+      reason: semantic highlighter fix for disk_bbq index options and knn with similarity
+
+  - do:
+      indices.create:
+        index: test-dense-index-bbq_disk
+        body:
+          settings:
+            index.mapping.semantic_text.use_legacy_format: false
```

For BWC tests we need to set this value to true

```suggestion
            index.mapping.semantic_text.use_legacy_format: true
```